### ISSUE TITLE
TestFX node lookup can cause javafx rendering issues

### ISSUE THREAD

**Issue by braway** [TestFX/TestFX#279]
We had random JavaFX rendering issues, like that nodes were shifted from their original position:
![radio-shift](https://cloud.githubusercontent.com/assets/1459982/15364434/2fab3856-1ce2-11e6-90bb-6177a2c8afe3.png)
(the second radio button is supposed to be below the upper one.)

After a lot of investigation, we found the issue:
when calling `FxRobot.clickOn(Node node, MouseButton... buttons)` from a Thread other than the FxApplicationThread, during Scene-layout, while the mouse is above the stage/scene that is being layouted, the layouting can be messed up.

Or in more simple words:
`node.getBoundsInLocal()` and/or `node.localToScene(Bounds localBounds)` can affect the layouting of a scene under certain circumstances, and should probably be performed on the FxApplicationThread.

I've created a little "test" ([StageTest.zip](https://github.com/TestFX/TestFX/files/270879/StageTest.zip)), which can be used to easily show the issue.
The test creates, asserts and hides a stage over and over again. The assertion code is partially copied from the `BoundsLocatorImpl`, to figure out if a node is within the bounds of the scene.

While the test is running, just move the mouse over the stage, and within a few seconds it should stop with an alert box. And a node should be misaligned or "missing":
![test-failure](https://cloud.githubusercontent.com/assets/1459982/15365630/9a56b87a-1ce6-11e6-876d-8c9171123713.png)
("Add" Button is missing, and Radio 3 label is misaligned)

tl;dr:
`node.localToScene(node.getBoundsInLocal())` (e.g in the `BoundsLocator`) should be performed on the FxApplicationThread or `FxRobot.clickOn(Node node, MouseButton... buttons)` (and others that use node lookup) should not be allowed on a non-FxApplicationThread.
